### PR TITLE
Rename behavior versions and associated functions

### PR DESCRIPTION
**Pull request by jdisanti**
The previous names are just too verbose. This PR shortens them and tries to make things clearer.

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

**Review by rcoh — APPROVED** [review]
looks good. please verify docs render as expected

File: design/src/SUMMARY.md [hunk 1]
```diff
@@ -66,7 +66,7 @@
   - [RFC-0037: The HTTP Wrapper](./rfcs/rfc0037_http_wrapper.md)
   - [RFC-0038: User-configurable retry classification](./rfcs/rfc0038_retry_classifier_customization.md)
   - [RFC-0039: Forward Compatible Errors](./rfcs/rfc0039_forward_compatible_errors.md)
-  - [RFC-0040: Behavior Major Versions](./rfcs/rfc0040_behavior_major_versions.md)
+  - [RFC-0040: Behavior Major Versions](./rfcs/rfc0040_behavior_versions.md)
```
Comment (ysaito1001) [inline design/src/SUMMARY.md:69]: I assume you deliberately left `RFC-0040: Behavior Major Versions` here as opposed to `RFC-0040: Behavior Versions` for a reason.

**Review by ysaito1001 — APPROVED** [review]
I personally like the new names `load_defaults` and `defaults`.

File: design/src/SUMMARY.md (hunk 1 above)
Comment (jdisanti) [reply design/src/SUMMARY.md:69]: Oops, I need to fix this one.

**Comment by jdisanti** [comment]
This is a breaking change as far as cargo-semver-checks is concerned, but it's not actually breaking since we haven't released yet.

**Pull request merged**